src/main/java/uk/gov/digital/ho/pttg/application/retry/NameMatchingCandidatesGenerator.java

```diff
@@ -12,45 +12,68 @@
+                    NamePairRule.create(2, 0), NamePairRule.create(2, 1),
+                    NamePairRule.create(0, 1), NamePairRule.create(1, 0)),
             4, ImmutableList.of(
-                    ImmutableList.of(0, 3), ImmutableList.of(1, 3), ImmutableList.of(2 ,3),
-                    ImmutableList.of(0, 1), ImmutableList.of(0, 2), ImmutableList.of(1, 0),
-                    ImmutableList.of(2, 0), ImmutableList.of(3, 0), ImmutableList.of(1, 2),
-                    ImmutableList.of(2, 1), ImmutableList.of(3, 1), ImmutableList.of(3, 2)),
+                    NamePairRule.create(0, 3), NamePairRule.create(1, 3), NamePairRule.create(2, 3),
+                    NamePairRule.create(0, 1), NamePairRule.create(0, 2), NamePairRule.create(1, 0),
+                    NamePairRule.create(2, 0), NamePairRule.create(3, 0), NamePairRule.create(1, 2),
+                    NamePairRule.create(2, 1), NamePairRule.create(3, 1), NamePairRule.create(3, 2)),
             5, ImmutableList.of(
-                    ImmutableList.of(0, 4), ImmutableList.of(1, 4), ImmutableList.of(2, 4), ImmutableList.of(3, 4),
-                    ImmutableList.of(0, 1), ImmutableList.of(0, 2), ImmutableList.of(0, 3), ImmutableList.of(1, 0),
-                    ImmutableList.of(1, 2), ImmutableList.of(1, 3), ImmutableList.of(2, 0), ImmutableList.of(2, 1),
-                    ImmutableList.of(2, 3), ImmutableList.of(3, 0), ImmutableList.of(3, 1), ImmutableList.of(3, 2),
-                    ImmutableList.of(4, 0), ImmutableList.of(4, 1), ImmutableList.of(4, 2), ImmutableList.of(4, 3))
+                    NamePairRule.create(0, 4), NamePairRule.create(1, 4), NamePairRule.create(2, 4), NamePairRule.create(3, 4),
+                    NamePairRule.create(0, 1), NamePairRule.create(0, 2), NamePairRule.create(0, 3), NamePairRule.create(1, 0),
+                    NamePairRule.create(1, 2), NamePairRule.create(1, 3), NamePairRule.create(2, 0), NamePairRule.create(2, 1),
+                    NamePairRule.create(2, 3), NamePairRule.create(3, 0), NamePairRule.create(3, 1), NamePairRule.create(3, 2),
+                    NamePairRule.create(4, 0), NamePairRule.create(4, 1), NamePairRule.create(4, 2), NamePairRule.create(4, 3))
     );
 
 
     public static List<String> generateCandidateNames(String firstName, String lastName) {
         validateName(firstName, lastName);
 
         List<String> candidates = new ArrayList<>();
-        List<String> allNames = findAllNames(firstName, lastName);
 
+        if(namesContainSplitters(firstName, lastName)) {
+            generateCandidatesWithSplitters(candidates, firstName, lastName);
+        }
+        else {
+            generateCandidates(candidates, firstName, lastName);
+        }
+
+        return candidates;
+    }
+
+    private static boolean namesContainSplitters(String firstName, String lastName) {
+        long firstNameSplitters = Arrays.stream(NAME_SPLITTERS.split("")).filter(c -> firstName.contains(c)).count();
```

Comment:
This method could be simplified by using `containsAny` from apache commons `StringUtils`

`isEmptyName` method below could also use `isBlank`

Reply:
Switched to StringUtils seeing as you are already using it in the project.  Not changed the arbitrary method name though because I prefer mine.